datalake/src/main/java/com/sequenceiq/datalake/flow/chain/DatalakeResizeRecoveryFlowEventChainFactory.java

```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.datalake.flow.chain;
+
+import static com.sequenceiq.datalake.flow.delete.SdxDeleteEvent.SDX_DELETE_EVENT;
+import static com.sequenceiq.datalake.flow.detach.SdxDetachEvent.SDX_DETACH_RECOVERY_EVENT;
+import static com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent.SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+import static com.sequenceiq.datalake.flow.start.SdxStartEvent.SDX_START_EVENT;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.flow.delete.event.SdxDeleteStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.DatalakeResizeRecoveryFlowChainStartEvent;
+import com.sequenceiq.datalake.flow.detach.event.SdxStartDetachRecoveryEvent;
+import com.sequenceiq.datalake.flow.start.event.SdxStartStartEvent;
+import com.sequenceiq.flow.core.chain.FlowEventChainFactory;
+import com.sequenceiq.flow.core.chain.config.FlowTriggerEventQueue;
+
+@Component
+public class DatalakeResizeRecoveryFlowEventChainFactory implements FlowEventChainFactory<DatalakeResizeRecoveryFlowChainStartEvent> {
+    @Override
+    public String initEvent() {
+        return SDX_RESIZE_RECOVERY_FLOW_CHAIN_START_EVENT;
+    }
+
+    @Override
+    public FlowTriggerEventQueue createFlowTriggerEventQueue(DatalakeResizeRecoveryFlowChainStartEvent event) {
+        Queue<Selectable> flowChain = new ConcurrentLinkedQueue<>();
+
+        // Forced deletion of the resized datalake.
+        boolean firstEventAdded = false;
+        if (event.getNewCluster().getDeleted() == null) {
+            flowChain.add(new SdxDeleteStartEvent(
```

Comment:
Can you please create a few methods in this class to instantiate these events. I think it would be cleaner to have methods like:
```java
SdxDeleteStartEvent deleteStartEvent(DatalakeResizeRecoveryFlowChainStartEvent event)

SdxStartDetachRecoveryEvent startDetachEvent(DatalakeResizeRecoveryFlowChainStartEvent event)

SdxStartStartEvent startDataLakeEvent(DatalakeResizeRecoveryFlowChainStartEvent event)
```

All those transformation logic can be hided this way.

Reply:
Done.